subsys/bluetooth/mesh/crypto.h

```diff
@@ -80,15 +83,30 @@ static inline int bt_mesh_dev_key(const uint8_t dhkey[32],
 int bt_mesh_prov_salt(const uint8_t conf_salt[16], const uint8_t prov_rand[16],
 		      const uint8_t dev_rand[16], uint8_t prov_salt[16]);
 
-int bt_mesh_net_obfuscate(uint8_t *pdu, uint32_t iv_index,
-			  const uint8_t privacy_key[16]);
+#define BT_MESH_NET_PRIVACY(r, iv) \
+	(uint8_t []) { 0, 0, 0, 0, 0, (iv) >> 24, (iv) >> 16, (iv) >> 8, (iv), \
+		       (r)[7], (r)[8], (r)[9], (r)[10], (r)[11], (r)[12], (r)[13] }
+
+int bt_mesh_net_obfuscate(const uint8_t *in, uint8_t *out,
+			  const uint8_t priv_plain[16], const uint8_t privacy_key[16]);
 
-int bt_mesh_net_encrypt(const uint8_t key[16], struct net_buf_simple *buf,
-			uint32_t iv_index, bool proxy);
+#define BT_MESH_NET_NONCE(n, iv, proxy) \
+	(uint8_t []) { (proxy) ? 3 : 0, (proxy) ? 0 : (n)[1], \
+		       (n)[2], (n)[3], (n)[4], (n)[5], (n)[6], 0, 0, \
+		       (iv) >> 24, (iv) >> 16, (iv) >> 8, (iv) }
```

Comment:
This is the main reason for the increased code space. I don't think it will affect the code execution speed?

Reply:
![image](https://user-images.githubusercontent.com/26378606/196398470-5d3371d2-2857-4fa3-b5f5-74434f867a58.png)

Reply:
I have also thought about not changing this method, but breaking AAPCS. Can also be packaged into functions, but it is necessary to define an external variable reference, which I think is not friendly.